src/package_staleness.py

```diff
@@ -0,0 +1,127 @@
+        # <version number>}, ..., {'url':<someurl>, 'dependencies': <List of dependencies>, 'version':
+        # <version number>}]
+        # We only care about the version number in the last index
+        package_version_in_conda = ''
+        if is_major_version_release:
+            package_version_in_conda = package_metadata[-1]['version']
+        elif is_minor_version_release:
+            package_major_version_prefix = versioning_split[0] + "."
+            package_version_in_conda = [x['version'] for x in package_metadata if
+                                        x['version'].startswith(package_major_version_prefix)][-1]
+        else:
+            package_minor_version_prefix = ".".join(versioning_split[:2]) + "."
+            package_version_in_conda = [x['version'] for x in package_metadata if
+                                        x['version'].startswith(package_minor_version_prefix)][-1]
+
+        package_to_version_mapping[package] = package_version_in_conda
+    return package_to_version_mapping
+
+
+def _generate_report(package_versions_in_conda_forge, package_versions_in_image, processor,
+                     version):
+    print('\n# Staleness Report: ' + version + '(' + processor + ')\n')
+    print('Package | Current Version in the Distribution image | Latest Relevant Version in '
+          'Conda Forge')
+    print('---|---|---')
+    for package in package_versions_in_conda_forge:
+        if package_versions_in_image[package] == package_versions_in_conda_forge[package]:
+            print(package + '|' + package_versions_in_image[package] + '|' +
+                  package_versions_in_conda_forge[package])
+        else:
+            print('${\color{red}' + package + '}$' + '|' + package_versions_in_image[package] +
+                  '|' + package_versions_in_conda_forge[package])
+
+
+def _get_installed_package_versions_and_conda_versions(processor, target_version_dir,
+                                                       target_version) \
+        -> (dict[str, str], dict[str, str]):
+    env_in_file_name = processor + '.env.in'
+    env_out_file_name = processor + '.env.out'
+    required_packages_from_target = get_required_packages_from_env_in(
+        target_version_dir + "/" + env_in_file_name
+    )
+    installed_packages_from_target, package_to_conda_forge_subdirectories_mapping = \
+        get_installed_packages_and_conda_forge_sub_dir_from_env_out(
+            target_version_dir + "/" + env_out_file_name)
+    # We only care about packages which are present in env.in
+    installed_packages_from_target = {k: v for k, v in installed_packages_from_target.items()
+                                      if k in required_packages_from_target}
+    # Remove Python from the dictionary, we don't want to track python version as part of our
+    # staleness report.
+    installed_packages_from_target.pop('python', None)
```

Comment:
We've tried hard to keep configs such as this in a small number of places.

Instead of adding `python` explicitly here, can we use the `_dependency_metadata` variable in the *dependency_upgrader.py` file?